File: backend/evidence-service/src/main/kotlin/com/evidentia/evidence/domain/Evidence.kt
package com.evidentia.evidence.domain

import com.evidentia.common.domain.TenantId
import java.time.Instant

data class Evidence(
    val id: EvidenceId,
    val tenantId: TenantId,
    val title: String,
    val description: String,
    val type: String,
    val sourceSystem: String,
    val owner: String,
    val approver: String? = null,
    val status: EvidenceStatus,
    val version: Int = 1,
    val createdAt: Instant,
    val updatedAt: Instant,
    val approvedAt: Instant? = null,
    val references: Map<String, String> = emptyMap(),
    val attachmentIds: List<String> = emptyList()
) {
    fun canTransitionTo(newStatus: EvidenceStatus): Boolean {
        return when (status) {
            EvidenceStatus.DRAFT -> newStatus == EvidenceStatus.IN_REVIEW
            EvidenceStatus.IN_REVIEW -> newStatus in listOf(EvidenceStatus.APPROVED, EvidenceStatus.REJECTED)
            EvidenceStatus.APPROVED -> newStatus == EvidenceStatus.LOCKED
            EvidenceStatus.REJECTED -> newStatus == EvidenceStatus.DRAFT // Can resubmit after rejection
            EvidenceStatus.LOCKED -> false // Locked evidence cannot be changed
        }
    }
    
    fun canBeEdited(): Boolean {
        return status in listOf(EvidenceStatus.DRAFT, EvidenceStatus.REJECTED)
    }
    
    fun canBeLocked(): Boolean {
        return status == EvidenceStatus.APPROVED
    }
    
    fun createNewVersion(): Evidence {
        return this.copy(
            version = version + 1,
            status = EvidenceStatus.DRAFT,
            approver = null,
            approvedAt = null,
            updatedAt = Instant.now()
        )
    }
}

File: frontend/compliance-portal/src/types/evidence.ts
import { z } from 'zod';

export const EvidenceStatusSchema = z.enum(['DRAFT', 'IN_REVIEW', 'APPROVED', 'REJECTED', 'LOCKED']);

export const EvidenceSchema = z.object({
  id: z.uuid(),
  tenantId: z.string(),
  title: z.string(),
  description: z.string(),
  type: z.string(),
  sourceSystem: z.string(),
  owner: z.string(),
  approver: z.string().nullable().optional(),
  status: EvidenceStatusSchema,
  version: z.number(),
  createdAt: z.string(),
  updatedAt: z.string(),
  approvedAt: z.string().nullable().optional(),
  references: z.record(z.string(), z.string()).default({}),
  attachmentIds: z.array(z.string()).default([]),
});

export const CreateEvidenceRequestSchema = z.object({
  title: z.string().min(1).max(500),
  description: z.string().min(1),
  type: z.string().min(1),
  sourceSystem: z.string().min(1),
  owner: z.string().min(1),
  references: z.record(z.string(), z.string()).optional().default({}),
});

export const UpdateEvidenceStatusRequestSchema = z.object({
  status: EvidenceStatusSchema,
});

export const ApiResponseSchema = <T extends z.ZodTypeAny>(dataSchema: T) =>
  z.object({
    success: z.boolean(),
    data: dataSchema.nullable(),
    error: z.object({
      code: z.string(),
      message: z.string(),
      details: z.record(z.string(), z.unknown()).optional().nullable(),
    }).nullable().optional(),
    timestamp: z.string(),
  });

export type Evidence = z.infer<typeof EvidenceSchema>;
export type CreateEvidenceRequest = z.infer<typeof CreateEvidenceRequestSchema>;
export type UpdateEvidenceStatusRequest = z.infer<typeof UpdateEvidenceStatusRequestSchema>;
